### ISSUE TITLE
Feature: repo page charts should be embeddable

### ISSUE THREAD

**Issue by jpmcb** [open-sauced/app#3860]
### Suggested solution

Lots of feedback we've gotten in the past is that charts should be embeddable. Our main use case for this we'll look at are contributors in the `ublue-os` org. There are 2 main scenarios we should target with this:

- embed directly in markdown: lots of projects _love_ to have these visualizations right in their GitHub readmes. Ideally, these should be able to be dropped into markdown somehow and render (even if they're static images that update once a day or so, that'd still be ideal)
  - Use case: https://github.com/ublue-os/bazzite - they embed star charts and contributor metrics directly in their readme. They'd love to use OpenSauced graphs from https://app.opensauced.pizza/s/ublue-os/bazzite (like the OpenSSF score, contributor confidence, etc.) - https://bazzite.gg/:
![Screenshot 2024-08-05 at 5 00 18 PM](https://github.com/user-attachments/assets/71acd166-2988-4e2e-a2a3-7446ebefd574)

- embed directly in sites: we've gotten feedback that people want to be able to drop these charts directly in their project websites.
  - Use case: Bazzite also has a project site that we've gotten signal would use our graphs directly in the site to show potential users or contributors that the project is "alive, well, and maintained" ![Screenshot 2024-08-05 at 5 03 06 PM](https://github.com/user-attachments/assets/42d80490-b24e-4270-a1b0-95514811428e)
---

- Big stretch goal: render in Slack or other third party electron based applications. There's lots of sharing and screen-shotting that happens on Slack: that's hard to capture on our end and we'd love the ability to render in these third party applications and redirect the user experience back to OpenSauced.

**Comment by nickytonline** [comment]
I had a chat with @brandonroberts about this yesterday. We're going to pair on it most likely.

The TLDR; of the POC

* Create either a Netlify function that uses Playwright
* Navigate via playwright to the page in question and use a selector that we'll add to find the graph.
* Wait until the data is rendered client-side
* Take a snapshot of that graph via Playwright
* Persist to Netlify Blob for quicker access on subsequent requests for sharing the image but cache for x amount of time

The function with the given parameters, e.g. /snapshot/denoland/deno?graph=lottery will be the share URL.

![CleanShot 2024-08-07 at 10 16 15](https://github.com/user-attachments/assets/f2dc087d-9fe9-4c4a-80d1-75c4f69324a5)


graph TD
    B[Create Netlify Function with Playwright]
    B --> C[Navigate to target page]
    C --> D[Use selector to find graph]
    D --> E[Wait for client-side data rendering]
    E --> F[Take snapshot of graph]
    F --> G[Persist snapshot to Netlify Blob and set cache duration]
    G --> H[Generate share URL]
    H --> I[End]

    A[Start] --> 
    K[User requests snapshot] --> L{Cached version available?}
    L -->|Yes| M[Return cached snapshot]
    L -->|No| B

    M --> I
    
    subgraph "Netlify Function"
    B
    C
    D
    E
    F
    G
    H
    I
    end
